mpris.c

```diff
@@ -283,6 +286,38 @@ static void try_put_youtube_thumbnail(GVariantDict *dict, char *path)
     g_match_info_free(match_info);
 }
 
+static void try_put_embedded_art(GVariantDict *dict, char *path)
+{
+    AVFormatContext *formatContext = NULL;
+    if (avformat_open_input(&formatContext, path, NULL, NULL) < 0) {
+        g_printerr("failed to open input file");
+        return;
+    }
+
+    if (avformat_find_stream_info(formatContext, NULL) < 0) {
+        g_printerr("failed to find stream info");
+        return;
+    }
+
+    AVPacket packet = {.size = 0};
+    for (unsigned int i = 0; i < formatContext->nb_streams; i++) {
+        if (formatContext->streams[i]->disposition & AV_DISPOSITION_ATTACHED_PIC) {
+            packet = formatContext->streams[i]->attached_pic;
+        }
+    }
+    if (!packet.size) {
+        return;
```

Comment:
Is another:
```cpp
avformat_close_input(&formatContext);
```
needed here?

Reply:
Nice catch, thanks. Rewrote the function.